src/python/pants/init/engine_initializer.py

```diff
@@ -86,18 +86,9 @@ def goal_consumed_subsystem_scopes(self, goal_name: str) -> Tuple[str, ...]:
 
     def goal_consumed_types(self, goal_product: Type) -> Set[Type]:
         """Return the set of types that could possibly be consumed while running the given goal."""
-        # TODO: This needs to be kept in sync with the Params injected in run_goal_rules, with a
-        # subtle difference that prevents automating validation that they are kept in sync:
```

Comment:
Huzzah!

> with a subtle difference that prevents automating validation

Now that this is no longer true, we could add an assert in `run_goal_rules` that this list matches? If not, maybe a comment keep the comment there at least.